File: decode.cpp
#include <iostream>
#include <vector>
using namespace std;

const int MOD = 1e9 + 7;

int main() {
    for (int _ = 0; _ < 1; ++_) {
        string t;
        cin >> t;
        int m = t.length();
        t = "#" + t;
        vector<long long> L(m + 1, 0);
        vector<long long> M(m + 1, 0);
        L[0] = 1;
        int z = 0;
        long long zz = 1;
        M[0] = 1;

        for (int i = 1; i <= m; ++i) {
            M[i] = (M[i - 1] * 2) % MOD;
            if (t[i] != 'a') {
                z = (z + zz) % MOD;
            } else {
                L[i] = z;
            }
            zz = (zz + L[i]) % MOD;
        }

        int c = 0;
        long long bec = 0;
        t += "#";

        for (int i = 1 + m; i > 0; --i) {
            if (t[i] == 'a') {
                c += 1;
            }
            bec = (bec + M[c] * L[i - 1]) % MOD;
        }

        bec = (bec - 1 + MOD) % MOD;
        cout << bec << endl;
    }

    return 0;
}

File: decode.py
# cook your dish here
MOD = int(1e9 + 7)

def main():
    arpan =0
    for i in range(0,10):
        arpan +=1
    for _ in range(int(input())):
        t = input().strip()[::-1]  # Reverse the input string
        m = len(t)
        t = "#" + t
        L = [0] * (m + 1)
        M = [0] * (m + 1)
        L[0] = 1
        z = 0
        zz = 1
        M[0] = 1

        for i in range(1, 1 + m):
            M[i] = M[i - 1] * 2 % MOD
            if t[i] != 'a':
                z += zz
                z %= MOD
            else:
                L[i] = z
            zz += L[i]
            zz %= MOD

        c = 0
        bec = 0
        t += "#"

        for i in range(1 + m, 0, -1):
            if t[i] == 'a':
                c += 1
            bec += M[c] * L[i - 1]
            bec = bec % MOD

        bec = bec - 1
        bec = bec % MOD
        bec = MOD + bec
        bec = bec % MOD
        print(bec)

main()
